### ISSUE TITLE
update-property chooses latest despite newVersion

### ISSUE THREAD

**Issue by metas-ts** [mojohaus/versions-maven-plugin#141]
I have the following output in my [jenkins job](https://jenkins.metasfresh.com/job/metasfresh-webui/job/master/2454/console). Login should work via your github account, in case you and to check for yourself.
The maven version is 3.3.9, the plugin version is 2.3.

I called mvn like this:
```
mvn --debug --settings /var/lib/jenkins/workspace/metasfresh-webui_master-HPHEUZSV5Q3XZYRC4FLCON4L3WXMLF6ALXY5SES6GGAYRHDALPKA@tmp/config3573879658530249706tmp --file pom.xml --batch-mode -Dtask-repo-id=metasfresh-task-repo -Dtask-repo-name=mvn-master -Dtask-repo-url=https://repo.metasfresh.com/content/repositories/mvn-master -Dproperty=metasfresh.version -DnewVersion=1.1781 versions:update-property
```

Here is an output that shows that is somehow ignores the 1.1781:

```
11:14:15 [DEBUG] Property ${metasfresh.version}: Set of valid available versions is [1.1781, 1.1782]
11:14:15 [DEBUG] Property ${metasfresh.version}: Restricting results to 1.1781
11:14:15 [DEBUG] Property ${metasfresh.version}: Current winner is: 1.1782
11:14:15 [DEBUG] Property ${metasfresh.version}: Searching reactor for a valid version...
11:14:15 [DEBUG] Property ${metasfresh.version}: Set of valid available versions from the reactor is []
11:14:15 [INFO] Updated ${metasfresh.version} from 1.0.0 to 1.1782
```

please ping me in case you need more info.

**Comment by khmarbaise** [comment]
Can you please give a test with the following:
```
mvn versions:update-property -Dproperty=... -DnewVersion=[1.1781]
```

**Comment by metas-ts** [comment]
thx for your reply. 
I tried it with
```
mvn versions:update-property -Dproperty=... -DnewVersion=[1.1793]
```

which got me
```
21:26:07 [DEBUG] Property ${metasfresh.version}: Set of valid available versions is [1.1781, 1.1782, 1.1786, 1.1787, 1.1793, 1.1794]
21:26:07 [DEBUG] Property ${metasfresh.version}: Restricting results to [1.1793,1.1793]
21:26:07 [DEBUG] Property ${metasfresh.version}: Current winner is: 1.1793
21:26:07 [DEBUG] Property ${metasfresh.version}: Searching reactor for a valid version...
21:26:07 [DEBUG] Property ${metasfresh.version}: Set of valid available versions from the reactor is []
21:26:07 [INFO] Updated ${metasfresh.version} from 1.0.0 to 1.1793
```

..so maybe this is not a bug but a matter of documentation (..or of *understanding* the documentation :blush: ).

**Comment by khmarbaise** [comment]
It is definitively a bug in documentation or lack of this...I'm already enhancing the documentation in that area...Thanks for finding this...
